src/readme.graph.md

```diff
@@ -534,6 +534,11 @@ directive:
         let duplicateDebugRegex = /^(\s*)(WriteDebug\(\$"{id}:.*)/gmi
         $ = $.replace(duplicateDebugRegex, "");
 
+        // catch all exceptions in ProcessRecordAsync.
+        let processAsyncFinallyRegex = /(finally\s*{\s*await \(\(Microsoft\.Graph\.PowerShell\.Runtime\.IEventListener\)this\)\.Signal\(Microsoft\.Graph\.PowerShell\.Runtime\.Events\.CmdletProcessRecordAsyncEnd\);)/gmi
+        let catchAllExceptionImplementation = '((Runtime.IEventListener)this).Signal(Runtime.Events.CmdletException, $"{ex.GetType().Name} - {ex.Message} : {ex.StackTrace}").Wait(); if (((Runtime.IEventListener)this).Token.IsCancellationRequested) { return; } WriteError(new global::System.Management.Automation.ErrorRecord(ex, string.Empty, global::System.Management.Automation.ErrorCategory.NotSpecified, null));'
```

Comment:
Is there a more specific error category than `ErrorCategory.NotSpecified`?

And for the Signal, a more specific exception than `Runtime.Events.CmdletException`?

Reply:
`NotSpecified` is best suited for this scenario since this is a catch-all uncaught exception. AutoREST already has a [feature request to map HTTP errors to PowerShell's error categories](https://github.com/microsoftgraph/msgraph-sdk-powershell/issues/274).

`CmdletException` is the only AutoREST runtime exception event that's available for us to listen to, see https://github.com/Azure/autorest.powershell/blob/main/powershell/resources/psruntime/EventsExtensions.cs